src/Iact1dUnbinnedLkl.cc

```diff
@@ -889,64 +889,78 @@ Int_t Iact1dUnbinnedLkl::ReaddNdESignal(TString filename)
   return status;
 }
 
-Int_t Iact1dUnbinnedLkl::ReaddNdESignal(TString* filenames, Float_t* branchingRatios, Int_t numFiles)
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+// 
+// Read multiple dN_i/dE for signal from file in
+// the Segue Stereo input format produced by Jelena
+// and their corresponding branching ratios BR_i.
+// Then perform a linear combination
+// dN/dE = sum(BR_i * dN_i/dE).
+// Replacement of existing file is allowed
+// Return 0 in case of success
+//        1 if file is not found
+//
+Int_t Iact1dUnbinnedLkl::ReaddNdESignal(Int_t nFiles, TString* filenames, Float_t* branchingRatios)
 {
-  // open file and look for histo
-  TFile* dNdESignalFile[numFiles];
-  TH1F*  hProvdNdESignal[numFiles];
+  // Initialize the array of pointers and other variables
+  TFile* dNdESignalFile[nFiles];
+  TH1F*  hProvdNdESignal[nFiles];
   TH1F*  linCHdNdESignal;
 
   Int_t status      = 0;
   Int_t nBins       = 0;
   Double_t lowEdge  = 0.0;
   Double_t highEdge = 0.0;
-  for(Int_t file=0;file<numFiles;file++)
+  // open file and look for histograms
+  for(Int_t iFile=0;iFile<nFiles;iFile++)
     {
-      dNdESignalFile[file]  = new TFile(filenames[file]);
-      hProvdNdESignal[file] = (TH1F*) dNdESignalFile[file]->Get("hdNdE");
-      if(!hProvdNdESignal[file]) status=1;
+      dNdESignalFile[iFile]  = new TFile(filenames[iFile]);
+      hProvdNdESignal[iFile] = (TH1F*) dNdESignalFile[iFile]->Get("hdNdE");
+      if(!hProvdNdESignal[iFile]) status=1;
       if(nBins!=0)
         {
-          if(hProvdNdESignal[file]->GetNbinsX()!=nBins) status=1;
-          if(hProvdNdESignal[file]->GetBinLowEdge(1)!=lowEdge) status=1;
-          if(hProvdNdESignal[file]->GetBinLowEdge(nBins+1)!=highEdge) status=1;
+          if(hProvdNdESignal[iFile]->GetNbinsX()!=nBins) status=1;
```

Comment:
This forces you to have all dNdE histograms with same binning, which may be true for the Cirelli distribution, but probably we don't want it to be a strict rule. I suggest you use the internal funcion readAndInterpolate to make sure all histograms have the same binning before you sume them

Reply:
Good point! Solved in the new commit!